module/zfs/vdev_raidz.c

```diff
@@ -2417,6 +2557,18 @@ vdev_raidz_io_start_read_row(zio_t *zio, raidz_row_t *rr, boolean_t forceparity)
 {
 	vdev_t *vd = zio->io_vd;
 
+	/*
+	 * Calculate how much parity is available for sitting out reads
+	 */
+	int parity_avail = rr->rr_firstdatacol;
+	for (int p = 0; p < rr->rr_firstdatacol; p++) {
+		raidz_col_t *rc = &rr->rr_col[p];
+		if (rc->rc_size > 0 &&
+		    !vdev_readable(vd->vdev_child[rc->rc_devidx])) {
+			parity_avail--;
+		}
+	}
```

Comment:
This does not take into account `vdev_dtl_contains()` below. Some of disks despite being readable might not have valid data.